flake.nix

```diff
@@ -0,0 +1,82 @@
+{
+  description =
+    "ripgrep, but also search in PDFs, E-Books, Office documents, zip, tar.gz, etc.";
+
+  inputs = {
+    nixpkgs-mozilla = {
+      url = "github:mozilla/nixpkgs-mozilla";
+      flake = false;
+    };
+    flake-utils.url = "github:numtide/flake-utils";
+    naersk.url = "github:nix-community/naersk";
+    pre-commit-hooks.url = "github:cachix/pre-commit-hooks.nix";
+  };
+
+  outputs =
+    { self, nixpkgs, nixpkgs-mozilla, flake-utils, naersk, pre-commit-hooks }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = (import nixpkgs) {
+          inherit system;
+
+          overlays = [ (import nixpkgs-mozilla) ];
+        };
+        toolchain = (pkgs.rustChannelOf {
+          rustToolchain = ./rust-toolchain;
+          sha256 = "sha256-u14sby2ZfEOIwPxfeL99++juoK6BCB9cSM5IFEkbKWw";
+        }).rust;
+        naersk' = pkgs.callPackage naersk {
+          cargo = toolchain;
+          rustc = toolchain;
+        };
```

Comment:
I'm planning to drop this and instead use the stable version of the Rust toolchain that's provided in the nixpkgs. As it currently stands, this SHA would require separate updates from regular `flake.lock` and isn't guaranteed to be in sync with that something like `rustup` would pull anyway.